packages/babel-parser/src/parser/expression.js

```diff
@@ -1227,13 +1229,27 @@ export default class ExpressionParser extends LValParser {
   }
 
   // https://github.com/tc39/proposal-do-expressions
-  parseDo(): N.DoExpression {
+  // https://github.com/tc39/proposal-async-do-expressions
+  parseDo(isAsync: boolean): N.DoExpression {
     this.expectPlugin("doExpressions");
+    if (isAsync) {
+      this.expectPlugin("asyncDoExpressions");
+    }
```

Comment:
We can throw a helpful error at https://github.com/babel/babel/blob/16d83002defc74dcbf0d73d8b71e9646acf04f16/packages/babel-parser/src/plugin-utils.js#L44 explaining that `asyncDoExpressions` can only be enabled when `doExpressions` is enabled.

Reply:
```js
if (hasPlugin(plugins, "asyncDoExpressions") && !hasPlugin(plugins, "doExpressions")) {
    throw new Error("The 'asyncDoExpressions' plugin also requires the 'doExpressions' plugin");
}
```

I guess not a problem when using babel-plugin-syntax-async-do-expressions since it adds more, but if you just do it manually?